src/lib/uri/uri_parser.rl

```diff
@@ -111,7 +111,9 @@ uri_raw_parse(struct uri_raw *uri, const char *p)
 
 		ip6	= ("[" (
 				((hex1_4?) ((":" (hex1_4?)){1,8})) |
-				("::" [ff][ff][ff][ff] ":" ip4addr))
+				("::" ip4addr) |
+				("::" [ff][ff][ff][ff] ":" ip4addr) |
+				("::" [FF][FF][FF][FF] ":" ip4addr))
```

Comment:
The IPv6 parsing in an URI should be case-insensitive according to [RFC 3986](https://datatracker.ietf.org/doc/html/rfc3986.html#section-3.2.2):

> A 128-bit IPv6 address is divided into eight 16-bit pieces. Each piece is represented numerically in **case-insensitive** hexadecimal <...>

It means that `::fFfF:192.168.1.1` should be accepted.

Reply:
corrected